src/Plot.js

```diff
@@ -414,18 +414,20 @@ export default class Plot extends Viz {
     let yBounds = this._yTest.outerBounds();
     let yWidth = yBounds.width ? yBounds.width + this._yTest.padding() : undefined;
 
-    this._y2Test
-      .domain(y2Exists ? y2Domain : yDomain)
-      .height(height)
-      .range([undefined, undefined])
-      .scale(y2Exists ? y2Scale.toLowerCase() : yScale.toLowerCase())
-      .select(testGroup.node())
-      .ticks(y2Ticks ? y2Ticks : yTicks)
-      .width(width)
-      .config(yC)
-      .config(defaultY2Config)
-      .config(this._y2Config)
-      .render();
+    if (y2Exists) {
+      this._y2Test
+        .domain(y2Exists ? y2Domain : yDomain)
+        .height(height)
+        .range([undefined, undefined])
+        .scale(y2Exists ? y2Scale.toLowerCase() : yScale.toLowerCase())
```

Comment:
if you're checking for `y2Exists` in the above `if` statement, then all of these methods don't need ternaries to check if `y2Exists` (it does if you're inside here!)